pkg/resources/digitalocean/dns/dns.go

```diff
@@ -0,0 +1,501 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package dns
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/digitalocean/godo"
+	"github.com/digitalocean/godo/context"
+
+	"k8s.io/kubernetes/federation/pkg/dnsprovider"
+	"k8s.io/kubernetes/federation/pkg/dnsprovider/rrstype"
+)
+
+const ipPlaceholder = "203.0.113.123"
+
+// DNS implements dnsprovider.Interface
+type DNS struct {
+	client *godo.Client
+}
+
+// NewProvider returns an implementation of dnsprovider.Interface
+func NewProvider(client *godo.Client) dnsprovider.Interface {
+	return &DNS{client: client}
+}
+
+// Zones returns an implementation of dnsprovider.Zones
+func (d *DNS) Zones() (dnsprovider.Zones, bool) {
+	return &zones{
+		client: d.client,
+	}, true
+}
+
+// zones is an implementation of dnsprovider.Zones
+type zones struct {
+	client *godo.Client
+}
+
+// List returns a list of all dns zones
+func (z *zones) List() ([]dnsprovider.Zone, error) {
+	domains, err := listDomains(z.client)
```

Comment:
In this case, you don't want to filter `List()` anyways since `zones.List()` is actually supposed to return all the zones registered under the cloud provider.